contracts/oracles/common/WrappedLiquidStakedTokenOracle.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity 0.8.13;
+
+import { OracleInterface } from "../../interfaces/OracleInterface.sol";
+import { ISFrax } from "../../interfaces/ISFrax.sol";
+import { ISfraxETH } from "../../interfaces/ISfraxETH.sol";
+import { ensureNonzeroAddress } from "@venusprotocol/solidity-utilities/contracts/validators.sol";
+import { EXP_SCALE } from "@venusprotocol/solidity-utilities/contracts/constants.sol";
+
+/**
+ * @title WrappedLiquidStakedTokenOracle
+ * @author Venus
+ * @notice This oracle fetches the price of wrapped liquid staked tokens
+ */
+abstract contract WrappedLiquidStakedTokenOracle is OracleInterface {
+    /// @notice A flag assuming 1:1 price equivalence between rebasing token and underlying token
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    bool public immutable ASSUME_EQUIVALENCE;
+
+    /// @notice Address of rebasing staked token
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable REBASE_TOKEN;
+
+    /// @notice Address of wrapped token obtained from rebasing rebasing token
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    address public immutable WRAPPED_TOKEN;
+
+    /// @notice Address of Resilient Oracle
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    OracleInterface public immutable RESILIENT_ORACLE;
+
+    /// @notice Address of WETH (on Ethereum chain) or ETH (on BNB chain)
```

Comment:
AFAIU, it could be also FraxETH... For me it's ok to use examples to clarify the potential values of the variables. I would do it for the three involved tokens. We can use the wstETH case:

* WRAPPED_TOKEN: wstETH
* REBASE_TOKEN: stETH
* UNDERLYING_TOKEN: WETH